tmt/beakerlib.py

```diff
@@ -55,17 +55,20 @@ class Library(object):
     workdir or into 'destination' if provided in the identifier.
     """
 
-    def __init__(self, identifier, parent=None):
+    def __init__(self, identifier, parent=None, gathered_hlp=[]):
```

Comment:
Do not use a mutable object as a default argument, see for example https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/ for explanation. Use `None` instead and then in the constructor `self.gathered_hlp = gathered_hlp or []`